### ISSUE TITLE
Migration runner skips v10–v15 on existing databases (recipients_json missing)

### ISSUE THREAD

**Issue by rdparker** [Dicklesworthstone/mcp_agent_mail_rust#50]
## Bug

The migration runner (`am migrate`) does not apply migrations v10 through v15 on an existing database that was created by an earlier version of the binary. This leaves the `messages` table missing the `recipients_json` column added in v15, causing `GET /mail` to fail with:

```json
{"detail":"SQLite error: Query error: no such column: m.recipients_json"}
```

## Steps to reproduce

1. Install `am` v0.2.7 or earlier — database is created with migrations through v9
2. Upgrade binary to v0.2.8 (or v0.2.9)
3. Run `am migrate` — output says:
   ```
   Database format: i64 microseconds (Rust native)
   ✓ Database schema created from model definitions!
   Note: To apply model changes, delete storage.sqlite3 and run this again.
   ```
4. Migrations v10–v15 are **not** applied; `recipients_json` column is missing
5. `GET /mail` returns the SQLite error above

## Expected behavior

`am migrate` should detect and apply all pending migrations (v10–v15) on an existing database, not just on a fresh one.

## Workaround

Manually add the column:

```sql
ALTER TABLE messages ADD COLUMN recipients_json TEXT NOT NULL DEFAULT '{}';
```

## Environment

- `am` v0.2.8 / v0.2.9
- Database originally created by v0.2.7 or earlier
- Linux x86_64
